packages/desktop-client/src/components/FinancesApp.js

```diff
@@ -172,78 +226,70 @@ function NavTab({ icon: TabIcon, name, path }) {
 }
 
 function MobileNavTabs() {
+  const { atLeastSmallWidth } = useResponsive();
   return (
     <div
       style={{
         backgroundColor: 'white',
         borderTop: `1px solid ${colors.n10}`,
         bottom: 0,
         ...styles.shadow,
-        display: 'flex',
+        display: atLeastSmallWidth ? 'none' : 'flex',
         height: '80px',
         justifyContent: 'space-around',
         paddingTop: 10,
         width: '100%',
       }}
     >
-      <NavTab name="Budget" path="/budget" icon={Wallet} isActive={false} />
-      <NavTab
-        name="Accounts"
-        path="/accounts"
-        icon={PiggyBank}
-        isActive={false}
-      />
-      <NavTab name="Settings" path="/settings" icon={Cog} isActive={false} />
+      <NavTab name="Budget" path="/budget" icon={Wallet} />
+      <NavTab name="Accounts" path="/accounts" icon={PiggyBank} />
+      <NavTab name="Settings" path="/settings" icon={Cog} />
     </div>
   );
 }
 
-class FinancesApp extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { isMobile: isMobile() };
-    this.history = createBrowserHistory();
+const patchedHistory = createBrowserHistory();
+
+function FinancesApp(props) {
+  const { atLeastSmallWidth } = useResponsive();
 
-    let oldPush = this.history.push;
-    this.history.push = (to, state) => {
+  useEffect(() => {
+    let oldPush = patchedHistory.push;
+    patchedHistory.push = (to, state) => {
```

Comment:
Since you can close and reopen files, should we be worried about this getting applied multiple times? (maybe you could store the history object in state so it gets recreated when this happens?

Reply:
It’s a good point.
 I haven’t fully grokked why we’re patching history anyway. I wonder if newer React Router versions would let us do this all differently.

Reply:
Hey @j-f1, I've been several approaches here, and while I can't find a material difference with the module-level var, I'm starting to think it's wise not to place the `useResponsive` hook this high in the render tree since it's triggered on every resize, which happens to solve the problem. The hook call was causing us to `createBrowserHistory` on every resize and as a result React Router complains "You cannot change <Router history>". Which was why I initially moved it to a module var.

Since my goal is not to change things here, just to refactor to a function component, I [pushed a change](5e6824246b5a1c50172d5495a780a55b3c3a35a7) so we DO `createBrowserHistory` on every render of FinancesApp, but we don't call `useResponsive` until a level down in the tree. It's maybe not _ideal_ to see <FloatableSidebar> and <Titlebar> in the JSX and not know immediately that they're not rendered at the narrowest widths, but now we create the history object fresh each time FinancesApp loads, just like on `master` today.

How's that strike you?